lib/linguist/heuristics.rb

```diff
@@ -89,6 +97,18 @@ def initialize(exts_and_langs, rules)
       @rules = rules
     end
 
+    # Internal: Return the heuristic's target extensions
+    def extensions
+      @exts_and_langs
+    end
+
+    # Internal: Return the heuristic's candidate languages
+    def languages
+      @rules.map do |rule|
+        [rule['language']].flatten(2).map { |name| Language[name] }
+      end.flatten.uniq
```

Comment:
Why are several `flatten` required here?

Reply:
Because [some](https://github.com/github/linguist/blob/96ad1185828f44bb9b774328a584551ee57ed264/lib/linguist/heuristics.yml#L292) heuristics return an array of candidates, so a second squash is needed to stamp out entries like `[Linux Kernel Module, AMPL]`.